optimum/habana/transformers/training_args.py

```diff
@@ -155,6 +161,21 @@ class GaudiTrainingArguments(TrainingArguments):
         },
     )
 
+    use_token_idx: bool = field(
+        default=False,
+        metadata={"help": "Whether to use token_idx for performing inference."},
+    )
+
+    use_kv_cache: bool = field(
+        default=True,
+        metadata={"help": "Use KV caching."},
+    )
+
+    ignore_eos: bool = field(
+        default=False,
+        metadata={"help": "Ignore eos token."},
+    )
+
```

Comment:
These should not be in the training arguments, let's use the argument `generation_config`: https://github.com/huggingface/optimum-habana/blob/1eef41c53ff60a12f6466c2775df03cbc0b3ebea/optimum/habana/transformers/training_args_seq2seq.py#L83

`use_cache` is already in Transformers' `GenerationConfig`: https://github.com/huggingface/transformers/blob/04ab5605fbb4ef207b10bf2772d88c53fc242e83/src/transformers/generation/configuration_utils.py#L38

Let's define a `GaudiGenerationConfig` to add `use_token_idx` and `ignore_eos`. I should probably do it because I'm not sure yet about how to do this properly.

Reply:
@regisss the default configuration we are using in Habana is use_token_idx :True and ignore_eos: False.
So, should I extend GenerationConfig to GaudiGenerationConfig and add use_token_idx and ignore_eos?
(or)
By "I'll try to have something merged in the next few days.", do you mean you'll get those changes merged?

Reply:
Yes, I think I'm going to do it because to be honest I'm still not sure of how exactly I would like to have this in the library. I'll have to think about it a little bit, but I'll do it in the next few days :slightly_smiling_face:

Reply:
We can remove all the modifications in this file.
`ignore_eos` will be automatically set to `True` if lazy mode is on, but you can explicitly set it to `False` in the generation config here: https://github.com/huggingface/optimum-habana/blob/7204bbbb1e28e11c3b99ff6e3b5a2bc9930886dc/examples/summarization/run_summarization.py#L677

Reply:
I rebased the changes on top of GaudiGenerationConfig. It's an involved change and I've responded in the comments for clarification. Meanwhile, I'm debugging issues related to accuracy as results are way off when I compare without GaudiGenerationConfig changes. 
Will push the new changes once I get it working with accuracy